Wangscape/tilegen/TilesetGenerator.cpp

```diff
@@ -13,8 +13,18 @@ namespace tilegen
 TilesetGenerator::TilesetGenerator(const Options& options,
                                    std::unique_ptr<partition::TilePartitionerBase> tile_partitioner) :
     options(options),
-    mTilePartitioner(std::move(tile_partitioner))
+    mTilePartitioner(std::move(tile_partitioner)),
+    mDebugTileBuilder(mDebugTile, {0., 0., 1., 1.})
+    // TODO (Serin) this sf::Rect is repeated in TilePartitionerNoise, factor it out
 {
+    if (options.debugOutput)
+    {
+        mDebugTile.create(options.tileFormat.resolution.x, options.tileFormat.resolution.y);
+        partition::TilePartitionerBase::DebugModuleWriter bound_debug_tile_writer = 
+            std::bind(&TilesetGenerator::writeDebugTile, this,
+                      std::placeholders::_1, std::placeholders::_2);
+        mTilePartitioner->setDebugModuleWriter(bound_debug_tile_writer);
```

Comment:
It doesn't seem to be a good idea. You're passing a pointer to this class' private function (which modifies `mDebugTilesets`) to an another class - it have to be fixed. As I suggested below, there could be a class, which generates debug tilesets and another one, which is responsible for writing tilesets to a given directory (probably it could also write tiles from `TilesetGenerator`.